### PR TITLE
Feature/asset bundle

### PR DESCRIPTION
**Pull request by keeramis**
## Description

Adds support to bundle assets in the CLI using the `particle bundle` command
### Usage
`particle bundle application.bin --assets /path/to/assets/folder --saveTo myBundle.zip`
- Providing `application.bin` is mandatory
- If `--assets` option is not specified, the files from the folder named `assets` in the current directory are picked up
    - :warning: A caveat with testing in particle-cli root directory is that there is an existing `assets` directory irrelevant to this implementation, and that might get picked up
- If `--saveTo` is not specified, default name is used

### Setup
1. Pull down the branch: `git pull && git checkout feature/asset-bundle`
2. Run `nvm use`
3. Reinstall dependencies: `npm run reinstall`
4. Run tests: `npm test`

### Creating a bundle
5. Save files of your choice to bundle in a folder (example: `/path/to/temp_assets/`
6. Specify the `application.bin` and the folder using the command 
```
particle-cli (feature/asset-bundle) $ npm start -- bundle /path/to/application.bin --assets /path/to/temp_assets/ --saveTo myBundle.zip
```
7. Verify that the bundle is created in the same directory
8. Verify from the CLI output that the expected files are included in the bundle

### Testing scenarios
1. Add special files like `__MACOS` or `.xyz` into the assets folder and verify that they are _not_ included in the bundle
2. Add nested directories in side the /assets folder and verify that they are _not_ included in the bundle
3. Please thoroughly test any additional customer-driven edge cases

## Related Issues / Discussions

[117747](https://app.shortcut.com/particle/story/117747/create-zipped-app-bundle)

## Completeness

- [x] User is totes amazing for contributing!
- [x] Contributor has signed [CLA](https://docs.google.com/a/particle.io/forms/d/1_2P-vRKGUFg5bmpcKLHO_qNZWGi5HKYnfrrkd-sbZoA/viewform)
- [x] Problem and solution clearly stated
- [x] Tests have been provided
- [x] Docs have been updated
- [] CI is passing

### CODE REVIEW

**Comment by keeramis** [comment]
I have observed instability in certain test e2e cases. Since this was on a linux-e2e / unix-e2e circle CI job, I am unable to reproduce this on my mac immediately. An example:
```
1) BundleCommands
       createBundle
         returns bundle with the default name if saveTo argument is not provided:
     Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/Users/distiller/project/src/cmd/bundle.test.js)
      at listOnTimeout (internal/timers.js:554:17)
      at processTimers (internal/timers.js:497:7)
```

File: .circleci/config.yml [hunk 1]
```diff
@@ -84,7 +84,18 @@ jobs:
       - run:
           name: Run Tests
           command: npm run test:ci
-  test-e2e:
+  test-e2e-linux:
```
Comment (keeramis) [inline .circleci/config.yml:87]: Once I 'touched' this file to remove node version `12` and include `16` for some of the jobs, circle CI started complaining of 
```
Job 'test-e2e-macos' requires 'test-e2e-linux', which is the name of 2 other jobs in workflow 'test-and-publish'
You can give a job within a workflow an explicit name by adding a `name` key
```
hence the change.
[link](https://app.circleci.com/pipelines/github/particle-iot/particle-cli/182)
